sdk/mobile-center-crashes/src/androidTest/java/com/microsoft/azure/mobile/crashes/CrashesAndroidTest.java

```diff
@@ -76,11 +77,25 @@ public void cleanup() {
     private void waitForCrashesHandlerTasksToComplete() throws InterruptedException {
         final Semaphore semaphore = new Semaphore(0);
 
-        /* Waiting background thread for initialize and processPendingErrors. */
-        Crashes.getInstance().mHandler.post(new Runnable() {
+        /* Waiting background thread for initialize. */
+        if (Crashes.getInstance().mCountDownLatchForLastSessionErrorReport != null)
+            Crashes.getInstance().mCountDownLatchForLastSessionErrorReport.await();
+
+        /* Waiting background thread for processPendingErrors. Wait for 2 second to make sure Crashes service starts processing pending errors. */
+        HandlerThread thread = new HandlerThread("TestThread");
```

Comment:
What I mean, is why HandlerThread? Seems like it's a silver bullet now, you could have used a single thread for example.